src/Our.Umbraco.DocTypeGridEditor/Web/Controllers/DocTypeGridEditorApiController.cs

```diff
@@ -85,5 +91,37 @@ public object GetDataTypePreValues(string dtdId)
             var propEditor = PropertyEditorResolver.Current.GetByAlias(dtd.PropertyEditorAlias);
             return propEditor.PreValueEditor.ConvertDbToEditor(propEditor.DefaultPreValues, preValue);
         }
+
+        [HttpPost]
+        public HttpResponseMessage GetPreviewMarkup([FromBody] FormDataCollection item, [FromUri] int nodeId)
+        {
+            // Get page container node
+            var page = UmbracoContext.ContentCache.GetById(nodeId);
+            if (page == null)
+            {
+                // If unpublished, then fake PublishedContent (with IContent object)
+                page = new UnpublishedContent(nodeId, Services);
+            }
+
+            var culture = UmbracoContext.Application.Services.ContentService.GetById(nodeId).GetCulture();
+            System.Threading.Thread.CurrentThread.CurrentCulture = culture;
+            System.Threading.Thread.CurrentThread.CurrentUICulture = culture;
```

Comment:
We'll need to clarify that this does only set the culture for the current thread, and not the application.
@mattbrailsford Do you have any insight on this?

Also, do we think this should also be in Stacked Content's previewer? (Although no one has mentioned culture issues to us).